### ISSUE TITLE
Training loss decreased and then increased

### ISSUE THREAD

**Issue by easonyang1996** [lucidrains/byol-pytorch#39]
Hi, I used your example on my own data. The training loss decreased and then increased after 100 epochs, which is wired. Did you meet similar situations? Is it hard to train the model?
the batchsize is 128/256
lr is 0.1/0.2
weight_decay is 1e-6

**Comment by easonyang1996** [comment]
I think the implementation in this repo is a little different to the byol model proposed in the original paper. In this repo, the weights of the target net is copied from the online network when the model is initialized. However, from the original paper, the online and target network have their own randomized initial weights at the beginning of model training. I wonder if it matters?

**Comment by lucidrains** [comment]
@easonyang1996 I don't think that's the issue, because as what Kaiming uncovered https://arxiv.org/abs/2011.10566 the networks can be the same, and a stop gradient on one of them is enough

**Comment by lucidrains** [comment]
@easonyang1996 in my example, I am using Adam, however, in the paper, they are using LARS. it could be that the learning rate you are using is specific to LARS. If you are using Adam, perhaps try a smaller learning rate?

**Comment by easonyang1996** [comment]
@lucidrains Thanks for your quick reply. You are right, I tried to use a smaller learning rate, the oscillating disappeared. Also thanks for the recommended paper!
